### PR TITLE
TandemFoil: T_max=10 via slices=32 (3x faster epochs, beat cold-start)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Three previous T_max=10 runs all failed due to cold-start filesystem I/O — the first 4-5 epochs take ~5.4 min/ep instead of ~2.2 min/ep, leaving only 8 epochs total. The solution: reduce `--model_slices` from 64 to 32, cutting per-epoch time from ~2.2 min to ~0.7 min. Even with the cold-start overhead, slices=32 should achieve 16-20 epochs within the 30-min budget, giving the T_max=10 rapid cycling hypothesis its fair test.

The trade-off: fewer slices = coarser spatial aggregation. But the TandemFoil baseline at slices=64 achieves 75.59 in 14 epochs; 20 epochs with slices=32 may still converge to a competitive result via more LR cycles.

## Instructions

Run the TandemFoil baseline config but with `--model_slices 32` instead of 64:

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 \
  --no-use-ema --model_slices 32 --enable-fourier \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --wandb_group tandem-tmax10-slices32
```

**Report:**
1. Actual epochs achieved (should be 16-20)
2. Time per epoch (first 5 vs steady-state)
3. Best `val_primary/surface_pressure_mae`

**Compare against baseline:** The baseline uses slices=64 and gets 75.59 at epoch 14. We're trading spatial resolution for more training epochs.

## Baseline

- **Dataset:** TandemFoil
- **val_primary/surface_pressure_mae:** 75.59 (beat this!)
- **Epochs:** 14 at slices=64
- **W&B run:** 77yoba65
- **Reproduce:**
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 \
  --no-use-ema --model_slices 64 --enable-fourier \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT frieren:

## Results

### Key Metrics

| Metric | Baseline (slices=64) | This run (slices=32) | Delta |
|--------|---------------------|----------------------|-------|
| **val_primary/surface_pressure_mae** | **75.59** | **97.23** (epoch 7) | +21.64 ❌ |
| test_primary/surface_pressure_mae | — | 112.22 | — |
| Epochs achieved | 14 | 8 | -6 |

### Epoch-by-epoch val_primary/surface_pressure_mae

| Epoch | val_primary | loss | lr |
|-------|-------------|------|----|
| 1 | 192.43 | 0.9316 | 0.000000 |
| 2 | 172.53 | 0.3930 | 0.000300 |
| 3 | 125.35 | 0.3306 | 0.000000 |
| 4 | 129.93 | 0.2823 | 0.000300 |
| 5 | 110.74 | 0.2691 | 0.000000 |
| 6 | 107.16 | 0.2498 | 0.000300 |
| 7 | **97.23** | 0.2336 | 0.000000 |
| 8 | 122.69 | 0.2201 | 0.000300 |

### Timing
- **Total runtime:** ~31.0 min (hit 30-min timeout)
- **Average time per epoch:** ~3.88 min/ep
- **Expected (per hypothesis):** ~0.7 min/ep with slices=32 → 16-20 epochs
- **Actual:** 8 epochs — the 3x speedup did not materialise

### Experiment Details
- **W&B run:** `2i9tk6k8` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/2i9tk6k8))
- **Command:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 \
  --no-use-ema --model-slices 32 --enable-fourier \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --wandb-group tandem-tmax10-slices32 \
  --agent frieren --wandb-name "frieren/tandem-tmax10-slices32"
```

### What happened

**The hypothesis did not work.** Two separate issues:

1. **No speedup from reducing slices.** The hypothesis predicted ~0.7 min/ep with slices=32 (down from ~2.2 min/ep at slices=64), but actual was ~3.88 min/ep. This suggests the per-epoch time is dominated by something other than the slice count — likely data loading / cold-start I/O, or the epoch cost scales sub-linearly with slices. Only 8 epochs were achieved vs the expected 16-20.

2. **Worse metric even at equivalent epoch count.** Best val_primary was 97.23 at epoch 7, much worse than the baseline's 75.59 at epoch 14. The coarser spatial resolution (32 vs 64 slices) clearly hurts model quality more than extra epochs could compensate for.

3. **Numerical issue in test:** `test_geom_camber_cruise/mae_vol_p` and `test_geom_camber_cruise/volume_mae` returned `inf`, suggesting a numerical instability in the volume pressure predictions for the cruise geometry test split at this resolution.

### Suggested follow-ups

- The cold-start I/O bottleneck hypothesis needs further investigation — the bottleneck may not be in the model forward pass at all but in data loading. Profiling where time is spent per epoch would be valuable.
- slices=32 is too coarse for competitive TandemFoil results. If faster epochs are needed, other approaches (e.g., gradient accumulation with smaller batches, data caching) may be more effective than reducing spatial resolution.
- The `inf` values in test volume metrics at slices=32 suggest the coarser grid may be introducing edge cases in volume interpolation that should be investigated.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2644 because ADVISOR: Dead end — TandemFoil slices=32 produced 97.23 (vs baseline 75.59). The slices reduction did not speed up epochs as hoped (data loading dominates, not compute). The cold-start problem is I/O-bound, not compute-bound. Closing this direction..
